cmd/manila-csi-plugin/main.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"strings"
+
+	"k8s.io/cloud-provider-openstack/pkg/csi/manila"
+	"k8s.io/klog"
+)
+
+var (
+	endpoint    = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	driverName  = flag.String("drivername", "manila.csi.openstack.org", "name of the driver")
+	nodeID      = flag.String("nodeid", "", "node id")
+	shareKind   = flag.String("sharekind", "", "kind of the Manila share along with the CSI endpoint for a dedicated CSI node plugin (expects a value in the format of SHARE-PROTO=UNIX-DOMAIN-SOCKET)")
```

Comment:
I wonder if we can think about a more specific name for this variable than ``shareKind``, maybe something along the lines of ``shareProtocolSelector``?  There are various ways of making "kinds" of shares in Manila so the very general name might confuse.

Reply:
Definitely! Accurate terminology is important, this one was just a wip name

Reply:
Done in https://github.com/kubernetes/cloud-provider-openstack/pull/536/commits/a773efe7cd37c552b85337124d07c46445939c2b